contextualized/regression/datasets.py

```diff
@@ -124,30 +119,33 @@ def _get_next_item(self, i):
         if self.n_i >= self.n:
             self.n_i = 0
             raise StopIteration
-        t = torch.zeros(self.x_dim + self.y_dim)
-        t[self.x_i] = 1
-        t[self.x_dim + self.y_i] = 1
+
+        t = tuple(
+            (
+                1 if i in (self.x_i, self.y_i) else 0
```

Comment:
I think this came up before as a suggestion to save on memory https://github.com/cnellington/Contextualized/pull/141#discussion_r1050901032, because they've actively being stored to self.sample_ids in init, unlike retrieving them from __get_item__

Reply:
You're absolutely right, sorry for the confusion. I didn't notice this was the indexing function. I believe I originally meant that we should store the task as a tuple (x_i, y_i), but now I realize that's a bit silly as well because we already return indices for index x_i and y_i. For the memory issue, we should use x_i and y_i to construct t in `__get_item__`